pkg/utils/maps/maps.go

```diff
@@ -34,3 +34,21 @@ func Merge(dest, src map[string]string) map[string]string {
 
 	return dest
 }
+
+// MergePreservingExistingKeys merges source into destination while skipping any keys that exist in the destination.
+func MergePreservingExistingKeys(dest, src map[string]string) map[string]string {
```

Comment:
Nit: I would have ordered the arguments as in the `cp`, `mv` or `ln` commands: `src` first, then `dest`.

Reply:
I was following the Go convention of using the destination first (e.g. [`copy()`](https://golang.org/pkg/builtin/#copy)) because I thought that would be more intuitive in this context. Happy to reconsider if you feel otherwise.

Reply:
I'm fine with the Go convention.